cmake/dev-helper.cmake

```diff
@@ -7,7 +7,7 @@ if(WITH_GALERA_DEV)
   ProcessorCount(CPU_COUNT)
 
   # Add a custom target for later refreshes
-  ADD_CUSTOM_TARGET(galera ALL scons -j ${CPU_COUNT} tests=0
+  ADD_CUSTOM_TARGET(galera ALL scons -j ${CPU_COUNT} debug=1 tests=0
```

Comment:
does this mean that it always builds debug?

Reply:
@kennt-percona  Yes. Without this, many galera MTR test cases skipped locally because of missing debug sync in galera.

Reply:
I guess my question is whether this should be dependent on the CMAKE_BUILD_TYPE or is debug always enabled (even for release builds).

Reply:
Good point! IIUC, this option was introduced to simplify the galera development for developers and I believe it is not being used in our Jenkins as well and we are the only ones who are using this option for building galera locally. So, as of now, I feel it is safe to build it in debug mode always.

But, given the advantages we get with building as per the CMAKE_BUILD_TYPE option, then definitely I agree with you. I will change the script accordingly and will update it.

Thank you for suggesting the improvement.